src/ql/pass/map/qubits/map/detail/mapper.cc

```diff
@@ -399,6 +399,14 @@ void Mapper::map_routed_gate(const ir::compat::GateRef &gate, Past &past) {
 
 }
 
+/**
+ * Nearest neighbors are in the same core, and at a distance of exactly one hop.
+ * This excludes the case of two communication qubits in different cores and at a distance of exactly one hop.
+ */
+utils::Bool Mapper::are_nearest_neighbors(UInt source, UInt target) const {
+    return platform->topology->get_min_hops(source, target) == 1;
```

Comment:
Looking again, get_min_hops is only used by the mapper and has to do with routing-specifics.
Maybe it's better to move get_min_hops out of topology and inside mapper. What do you think?

Reply:
I'm with Hans here in that there may be coming changes to Topology in a near future. I would rather keep things as they are at the moment. Even if they changes weren't coming, I would indeed separate this comment's proposal to a different commit. So, let's say, in any case, I would limit the scope of this commit to just fixing the distance cases.